iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -1265,6 +1277,19 @@ static int setMqttMessagePropertyIfPossible(IOTHUB_MESSAGE_HANDLE IoTHubMessage,
     // Not finding a system property to map to isn't an error.
     int result = 0;
 
+    if (nameLen > 5)
```

Comment:
It would be good to define what 5 is at the top (as a macro), so it doesn't become a magic number.

Reply:
It's not just this 5, you would need to refactor the function to not make this a one off.